### ISSUE TITLE
Workflow.Await, Workflow.Sleep to show custom message/additional context about the await/sleep event in history

### ISSUE THREAD

**Issue by kkcmadhu** [temporalio/web#357]
**Is your feature request related to a problem? Please describe.**
When workflow waits on a condition or goes for a sleep , the history in tctl and web shows timer/ sleep activity. but will will be nice if a custom message can be added to give hints on why the workflow went to await or sleep

**Describe the solution you'd like**
Api/sdk enhancement which supports something like
Workflow.await(maxDuration, () -> getApproval(), “waiting for admin Approval”);
Workflow.sleep(Duration.days(2), “waiting for cool of period to get over”);

**Describe alternatives you've considered**

**Additional context**
Workflow history shows the signal names, acticity names, child workflow names, but for workflow.sleep and workflow.await it shows up as Timer'SomeRandomNumber()' , instead custom message could be very meaningful.

**Comment by samarabbas** [comment]
This seems to an ask for improved TemporalWeb experience.  Moved under TemporalWeb repo.

**Comment by madhukkc** [comment]
yes in general the ui, but the  Workflow.sleep() api may also need improvment to take a event name, because unlike activity/signal there is no implcit name attached to it..

**Comment by p4p4** [comment]
I think it would require changes in
* the server (to be able to also append "messages" to the history 
* the SDKs (to allow passing a message, either as part of await, or as a standalone Workflow.appendMessage("waiting for foo") method. 
* the UI (to show them in the history properly)

Would really appreciate such a feature. In the meantime, I'd probably create a "dummy activity" just to log in the Worfklow:

in the workflow:
```
activity.logMessage("waiting for signal");
Workflow.await(() -> this.isSignalReceived)
```

and an activity:
```
// WorkflowMethod
String logMessage(String message) {
 return message;
}
```


but actually it would be nicer if we could just write
```
Workflow.await(() -> this.isSignalReceived, "waiting for signal")
```
